### ISSUE TITLE
StackFrame.GetFileLineNumber() works in netcoreapp1.1 but not netcoreapp2.0

### ISSUE THREAD

**Issue by NTaylorMullen** [dotnet/coreclr#11378]
In netcoreapp1.1 it returns the correct line, in netcoreapp2.0 it returns 0.

Code to show the error (in the repro app included at the end):
```C#
Exception ex;

try
{
    throw new Exception("Hello");
}
catch(Exception e)
{
    ex = e;
}

var needFileInfo = true;
var stackTrace = new System.Diagnostics.StackTrace(ex, needFileInfo);
var stackFrames = stackTrace.GetFrames();

foreach (var frame in stackFrames)
{
    Console.Write(frame.GetMethod().Name + " : ");
    Console.WriteLine(frame.GetFileLineNumber()); // This will always write out 0 in netcoreapp2.0
}
```

Repro app: 
[StackTraceIssue.zip](https://github.com/dotnet/coreclr/files/974972/StackTraceIssue.zip)

Try running the app with both TFMs to see the difference.

/cc @Eilon @DamianEdwards

**Comment by DamianEdwards** [comment]
This is impacting ASP.NET Core in that our developer exception page (the page shown to developers in the browser when an exception is thrown in their application) no longer contains the line number in their own code where the exception occurred. As you can imagine this makes finding the line of code that's throwing (especially with async stacks) much harder.

**Comment by Petermarcu** [comment]
@brianrob can  you take a look?

**Comment by mikem8361** [comment]
If you are using portable PDB's then the reason is issue #19319. If on Windows and Windows PDB's then I'll need to investigate a little (mainly to see if microsoft.diasymreader.native.amd64.dll is exists).

**Comment by lt72** [comment]
Mike is referring to issue [19319](https://github.com/dotnet/corefx/issues/19319) in CoreFX repo.

**Comment by mikem8361** [comment]
Fixed PR #19364
